src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksAchServiceImpl.java

```diff
@@ -0,0 +1,299 @@
+			String closer = "'";
+			for (String key : fieldChanges.keySet()) {
+				LOG.debug(new StringBuilder(keyAndStarterString).append(key).append(valueString).append(key).append(closer).toString());
+			}
+		}
+	}
+
+	@Override
+	public PayeeACHAccount createPayeeAchAccount(PayeeACHAccount payeeAchAccountOld, String routingNumber, String accountNumber) {
+		PayeeACHAccount payeeAchAccount = (PayeeACHAccount) ObjectUtils.deepCopy(payeeAchAccountOld);
+		payeeAchAccount.setBankRoutingNumber(StringUtils.defaultIfEmpty(routingNumber, payeeAchAccount.getBankRoutingNumber()));
+		payeeAchAccount.setBankAccountNumber(StringUtils.defaultIfEmpty(accountNumber, payeeAchAccount.getBankAccountNumber()));
+		payeeAchAccount.setObjectId(null);
+		payeeAchAccount.setVersionNumber(null);
+		payeeAchAccount.setAchAccountGeneratedIdentifier(null);
+		payeeAchAccount.setActive(true);
+		payeeAchAccount.setLastUpdateUserId(PaymentWorksConstants.SOURCE_USER);
+		payeeAchAccount.setLastUpdate(getDateTimeService().getCurrentTimestamp());
+		return payeeAchAccount;
+	}
+	
+	@Override
+	public boolean processACHUpdates(List<PaymentWorksVendorUpdatesDTO> achUpdates, PaymentWorksVendorUpdateResults resultsDTO) {
+		boolean routed = false;
+		for (PaymentWorksVendorUpdatesDTO vendorUpdate : achUpdates) {
+			LOG.info("processACHUpdates, processing " + vendorUpdate.getVendor_name() + " with an ID of " + vendorUpdate.getId());
+			if (!getPaymentWorksVendorService().isExistingPaymentWorksVendor(vendorUpdate.getId(), PaymentWorksConstants.TransactionType.ACH_UPDATE)) {
+				LOG.info("processACHUpdates is NOT an existing payment works vendor request for " + vendorUpdate.getId() + ", will process");
+				routed = processSingleACHUpdate(vendorUpdate, resultsDTO) && routed;
+			} else {
+				LOG.info("processACHUpdates, There is an existing request with the ID of " + vendorUpdate.getId() + " so won't process");
+			}
+		}
+		return routed;
+	}
+
+	protected boolean processSingleACHUpdate(PaymentWorksVendorUpdatesDTO vendorUpdate, PaymentWorksVendorUpdateResults resultsDTO) {
+		boolean routed;
+		PaymentWorksVendor paymentWorksVendor = getPaymentWorksVendorService().savePaymentWorksVendorRecord(
+				vendorUpdate, PaymentWorksConstants.ProcessStatus.ACH_UPDATE_COMPLETE, PaymentWorksConstants.TransactionType.ACH_UPDATE);
+		
+		try {
+			if (ObjectUtils.isNotNull(vendorUpdate.getField_changes().getField_changes())) {
+				routed = getPaymentWorksKfsService().directAchEdit(vendorUpdate, paymentWorksVendor.getVendorNumberList());
+			} else {
+				throw new ValidationException("No field changes provided by PaymentWorks");
+			}
+		} catch (Exception e) {
+			routed = false;
+			resultsDTO.setHasErrors(true);
+			GlobalVariables.getMessageMap().clearErrorMessages();
+			GlobalVariables.getMessageMap().clearErrorPath();
+			LOG.error("processSingleACHUpdate, There was error processing vendor update with ID of " + vendorUpdate.getId() + " because " + e.getMessage());
```

Comment:
If you would rather log the full stack trace and not just the exception message, you could pass the exception into the LOG.error method as a second argument.